src/averageAges.js

```diff
@@ -55,7 +68,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(person => person.sex === 'm'
+    && people.some(mother => mother.name === person.mother))
+    : people.filter(person => people.some(mother =>
+      mother.name === person.mother));
```

Comment:
You have repeated logic in two parts of your comparison.
Try to improve the readability of this code.

Reply:
Review @33Orange comment. You can have a check for `onlyWithSon` inside a first `filter`, then second filter won't be needed